helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvscWriter.java

```diff
@@ -173,20 +249,63 @@ protected void toJson(
                 } else {
                     gen.writeStartObject();
                     gen.writeStringField("type", schema.getName());
-                    writeProps(schema, gen);
+                    Set<String> claimedProps = executeAvscWriterPluginsForSchema(schema, gen);
+
+                    if(retainNonClaimedProps) {
+                        writeProps(schema, gen, claimedProps);
+                    }
                     gen.writeEndObject();
                 }
         }
     }
 
-    protected void aliasesToJson(Schema schema, List<AvroName> extraAliases, G gen) throws IOException {
-        Set<String> userDefinedAliases = schema.getAliases();
-        Set<String> allAliases = userDefinedAliases; //could be null
-        if (addAliasesForAvro702 && extraAliases != null) {
-            allAliases = new HashSet<>();
-            if (userDefinedAliases != null) {
-                allAliases.addAll(userDefinedAliases);
+    /***
+     * Runs execute(Schema, JsonGeneratorWrapper) for provided AvscWriterPlugins
+     * @param schema
+     * @param gen
+     * @return list of Json prop names handled by plugins
+     * @throws UnsupportedOperationException of 2+ plugins provided for same json prop
+     */
+    private Set<String> executeAvscWriterPluginsForSchema(Schema schema, G gen) {
+        Set<String> claimedProps = new HashSet<>();
+        for(AvscWriterPlugin plugin : _plugins) {
+            String propName = plugin.execute(schema, gen);
+            if(propName!= null && !propName.isEmpty()) {
+                if(claimedProps.contains(propName)) {
+                    throw new UnsupportedOperationException("Only 1 AvscWriterPlugin per Json property allowed.");
```

Comment:
you can make this error more actionable if claimedProp is a Map<String, Plugin> - that way you could tell which 2 plugins claim the same prop (the existing map value and the one currently being run)

Reply:
This approach will limit the error to 2 plugins per propName. Wdyt about "Multiple AvscWriterPlugin(s) defined for {propname}. Only 1 plugin per Json property allowed."